packages/go_router/lib/src/route.dart

```diff
@@ -433,6 +442,7 @@ class ShellRoute extends RouteBase {
   ShellRoute({
     this.builder,
     this.pageBuilder,
+    this.nestedNavigationBuilder,
     super.routes,
     GlobalKey<NavigatorState>? navigatorKey,
```

Comment:
It feels weird that we still allow navigator key to be passed in if nestedNavigationBuilder is used, but it won't actually be used to build a navigator. We may need a less confusing way to match subpages to the shell route if nestedNavigationBuilder is in used.

Reply:
We can also consider a different named constructor to limit/change the constructor parameters

Reply:
Yes, a `ShellRoute.nested` named constructor would fit quite well here.

Reply:
Great input, thanks! As mentioned below, this is now solved with a new shell route class (PartitionedShellRoute).